tests/PHPStan/Rules/Comparison/data/bug-4570.php

```diff
@@ -0,0 +1,52 @@
+<?php declare(strict_types = 1);
+
+namespace Bug4570;
+
+use function PHPStan\Testing\assertType;
+
+trait MyLogic
+{
+	public function fetchValue() : string {
+		assertType('string', self::MY_CONST);
```

Comment:
How can this assert worj when the FirstConsumer class has `MY_CONST = 1;`?

Reply:
asserting types when using a trait form 2 classes is kind of hard to express. I therefore moved the type assertions into a separate case which only uses a single class

Reply:
Bu that kind of defeats the purpose of this fix. When one class has `const MY_CONST = 1;` and another one `const MY_CONST = 'foo';`, PHPStan is still going to complain.